### PR TITLE
Add InitialApplier, LastApplier and LastAppliedAt to Campaign

### PR DESCRIPTION
**Pull request by mrnugget**
This fixes https://github.com/sourcegraph/sourcegraph/issues/12467 by
recording these fields and exposing them in the API.

### CODE REVIEW

File: enterprise/internal/campaigns/resolvers/campaigns.go [hunk 1]
```diff
@@ -102,12 +102,30 @@ func (r *campaignResolver) Branch() *string {
...
+	return graphqlbackend.UserByIDInt32(ctx, r.Campaign.LastApplierID)
+}
+
+func (r *campaignResolver) LastAppliedAt() graphqlbackend.DateTime {
+	return graphqlbackend.DateTime{Time: r.Campaign.LastAppliedAt}
+}
+
+func (r *campaignResolver) SpecCreator(ctx context.Context) (*graphqlbackend.UserResolver, error) {
+	spec, err := r.store.GetCampaignSpec(ctx, ee.GetCampaignSpecOpts{
+		ID: r.Campaign.CampaignSpecID,
```
Comment (eseliger) [inline enterprise/internal/campaigns/resolvers/campaigns.go:119]: Do migrated campaigns have one? Would it error for existing ones now?

File: enterprise/internal/campaigns/service.go [hunk 2]
```diff
@@ -314,11 +314,17 @@ func (s *Service) ApplyCampaign(ctx context.Context, opts ApplyCampaignOpts) (ca
...
-	campaign.AuthorID = campaignSpec.UserID
 	campaign.NamespaceOrgID = campaignSpec.NamespaceOrgID
 	campaign.NamespaceUserID = campaignSpec.NamespaceUserID
 	campaign.Name = campaignSpec.Spec.Name
 
+	actor := actor.FromContext(ctx)
+	if campaign.InitialApplierID == 0 {
+		campaign.InitialApplierID = actor.UID
+	}
+	campaign.LastApplierID = actor.UID
```
Comment (eseliger) [inline enterprise/internal/campaigns/service.go:325]: Doesn't need to be nullable in the DB then, right?

File: enterprise/internal/campaigns/service.go [hunk 3]
```diff
@@ -314,11 +314,17 @@ func (s *Service) ApplyCampaign(ctx context.Context, opts ApplyCampaignOpts) (ca
...
+	campaign.LastAppliedAt = s.clock()
```
Comment (eseliger) [inline enterprise/internal/campaigns/service.go:326]: ..and so does this field.

File: enterprise/internal/campaigns/resolvers/campaigns.go (hunk 1 above)
Comment (mrnugget) [reply enterprise/internal/campaigns/resolvers/campaigns.go:119]: We're doing to delete those. But good point, I could add a check here and return nil.

File: enterprise/internal/campaigns/service.go (hunk 2 above)
Comment (mrnugget) [reply enterprise/internal/campaigns/service.go:325]: Yeah, technically. But I think nullable fields make testing easier. Unsure... @LawnGnome thoughts?

Comment (mrnugget) [reply enterprise/internal/campaigns/service.go:325]: I think one approach might be to first write the migration and _then_ make the fields not nullable.

**Pull request merged**